### PR TITLE
feat(examples): Adding Tracetest Typescript Example

### PR DESCRIPTION
**Pull request by xoscar**
This PR updates the examples folder adding the Typescript example using the new `@tracetest/client` package.

## Changes

- Adds new example
- Adds new recipe

## Fixes

- https://github.com/kubeshop/tracetest-cloud/issues/392

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Loom video
localhost_3000_tools-and-integrations_typescript

### CODE REVIEW

File: examples/quick-start-typescript/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,168 @@
+version: '3.5'
+name: pokeshop
+
+services:
+  app:
```
Comment (adnanrahic) [inline examples/quick-start-typescript/docker-compose.yaml:5]: I'd say naming this `tracetest-client` makes most sense.

File: examples/quick-start-typescript/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      cache:
+        condition: service_healthy
+      queue:
+        condition: service_healthy
+      jaeger:
+        condition: service_healthy
+      otel-collector:
+        condition: service_started
+
+  tracetest-agent:
```
Comment (adnanrahic) [inline examples/quick-start-typescript/docker-compose.yaml:161]: Putting this at the top with `tracetest-client` would make sense to keep them together for clarity.

File: examples/quick-start-typescript/index.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import Tracetest from '@tracetest/core';
```
Comment (adnanrahic) [inline examples/quick-start-typescript/index.ts:1]: I think renaming `@tracetest/core` to `@tracetest/client` will make sense in the long run. We would only call the OSS product Tracetest Core. WDYT?

**Comment by adnanrahic** [comment]
This is awesome! Great work @xoscar 💪 
We're moving into the space of programmatically triggering tests and it's super cool.

I have a technical question. How does the npm client library interact with Tracetest Agent? I don't quite understand that part.

File: examples/quick-start-typescript/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Tracetest + Typescript (using @tracetest/client NPM Package)
+
+> [Read the detailed recipe for setting up Tracetest + Typescript (using @tracetest/client NPM Package) in our documentation.](https://docs.tracetest.io/tools-and-integrations/typescript)
+
+This repository's objective is to show how you can run trace-based tests from your Javascript/Typescript environment, including setup stages and waiting for results to be ready.
+
+## Steps
+
+1. Copy the `.env.template` file to `.env`.
+2. Login-into the [Tracetest app](https://app.tracetest.io/).
```
Comment (jfermi) [inline examples/quick-start-typescript/README.md:10]:
```suggestion
2. Log into the [Tracetest app](https://app.tracetest.io/).
```

**Review by jfermi — COMMENTED** [review]
An editing suggestion.

**Review by kdhamric — APPROVED** [review]

**Pull request merged**